File: engine/model/src/main/scala/tools/packet/npctalk/GetNPCTalk.scala
package tools.packet.npctalk

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GetNPCTalk(private var _npcId: Int, private var _messageType: Byte, private var _talk: String,
                 private var _endBytes: String, private var _speaker: Byte) extends PacketInput {
  def npcId: Int = _npcId

  def messageType: Byte = _messageType

  def talk: String = _talk

  def endBytes: String = _endBytes

  def speaker: Byte = _speaker

  override def opcode(): SendOpcode = SendOpcode.NPC_TALK
}

File: engine/engine-model/src/main/java/tools/packet/npctalk/GetNPCTalk.java
package tools.packet.npctalk;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GetNPCTalk(Integer npcId, Byte messageType, String talk, String endBytes,
                         Byte speaker) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.NPC_TALK;
   }
}